datafusion/core/src/execution/context/mod.rs

```diff
@@ -1277,6 +1297,11 @@ impl SessionContext {
         state
     }
 
+    /// Get reference to [`SessionState`]
+    pub fn state_ref(&self) -> Arc<RwLock<SessionState>> {
```

Comment:
This is not directly related to the issue we are resolving, but I found it useful for the user to get the `SessionState` reference. So this method is added as a complement to `state()` and `state_weak_ref()` that we already have.